plugins/OpenDeck/OpenDeckPlugin.cpp

```diff
@@ -38,58 +38,62 @@ constexpr char leftWallBufferName[] = "0L";
 constexpr char rightWallBufferName[] = "0R";
 constexpr char leftFloorBufferName[] = "1L";
 constexpr char rightFloorBufferName[] = "1R";
+
+const std::string CAMERA_TYPE = "cylindricStereoTracked";
+const std::string CAMERA_SCALING_PROP = "cameraScaling";
 }
 
-OpenDeckPlugin::OpenDeckPlugin(const Vector2ui& wallRes,
-                               const Vector2ui& floorRes)
-    : _wallRes(wallRes)
-    , _floorRes(floorRes)
+OpenDeckPlugin::OpenDeckPlugin(OpenDeckParameters&& params)
+    : _params(std::move(params))
 {
+    if (_params.getResolutionScaling() > 1.0f ||
+        _params.getResolutionScaling() <= 0.0f)
+    {
+        throw std::runtime_error(
```

Comment:
that has to be catched in the C create() function, no exceptions are allowed to pass through there